site/static/api/device-registry-revised.yaml

```diff
@@ -0,0 +1,734 @@
+
+      get:
+         tags:
+            - credentials
+         summary: Get credentials set of a device.
+         description: |
+            Get the credentials set of a device. As long as the device is
+            registered and the user has read access to is, this call should
+            never return "not found".
+         operationId: getAllCredentials
+         responses:
+            200:
+               description: Operation successful
+               content:
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/CredentialsSet'
+               headers:
+                  ETag:
+                     description: Version of the resource
+                     schema:
+                        type: string
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            404:
+               $ref: '#/components/responses/NotFound'
+
+      put:
+         tags:
+            - credentials
+         summary: Update credentials set for registered device
+         operationId: setAllCredentials
+         parameters:
+            - $ref: '#/components/parameters/resourceVersion'
+         requestBody:
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/CredentialsSet'
+            required: true
+         responses:
+            204:
+               $ref: '#/components/responses/Updated'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+      delete:
+         tags:
+            - credentials
+         summary: Delete all credentials of a device
+         description: |
+             Deletes all credentials from a device registration. The device is
+             still registered, and a following call to *get* credentials must
+             not return `404`, as the device is still registered. Instead an
+             empty result must be returned.
+         operationId: deleteAllCredentials
+         parameters:
+            - $ref: '#/components/parameters/resourceVersion'
+         responses:
+            204:
+               $ref: '#/components/responses/Delete'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+components:
+
+   schemas:
+
+# Common schema
+
+      Error:
```

Comment:
Since you already define a specific error response object, I think it would be good to include a (optional) error-code field. A frontend might use it to show localized messages or an automated client might use it for a more specific errorhandling.

Reply:
@wistefan which partsdid you implement ? I wanted to start this work during this week but there is no need to duplicate the effort if you've worked on it. Would you like some help ?

Reply:
The whole Rest-Part of the api, including persistence in a sql-database. Since I implemented it using micronaut I'm not totally sure if it will fit into the whole hono project and I still have some work to do until I can put it on github(f.e. resolving the dependencies to our internal openapi-generator). It will probably take a couple of weeks(~4) until I can prepare a PR.

Reply:
So you did not update the `HttpEndpoint` classes ?

Reply:
Correct.